File: backend/app/schemas/analytics.py
from decimal import Decimal
from pydantic import BaseModel


class CategorySpend(BaseModel):
    category: str
    amount: Decimal
    count: int
    percentage: float


class MonthlySpend(BaseModel):
    month: str  # "2026-01"
    amount: Decimal


class AnalyticsResponse(BaseModel):
    total_spent: Decimal
    this_month: Decimal
    last_month: Decimal
    net_balance: Decimal
    group_count: int
    by_category: list[CategorySpend]
    by_month: list[MonthlySpend]

File: app/src/main/java/com/expensio/domain/model/Analytics.kt
package com.expensio.domain.model

import java.math.BigDecimal

data class CategorySpend(
    val category: String,
    val amount: BigDecimal,
    val count: Int,
    val percentage: Float,
)

data class MonthlySpend(
    val month: String,   // "2026-01"
    val amount: BigDecimal,
)

data class Analytics(
    val totalSpent: BigDecimal,
    val thisMonth: BigDecimal,
    val lastMonth: BigDecimal,
    val netBalance: BigDecimal,
    val groupCount: Int,
    val byCategory: List<CategorySpend>,
    val byMonth: List<MonthlySpend>,
)
